BonusCalcApi.Tests/DatabaseTests.cs

```diff
@@ -14,9 +16,18 @@ public abstract class DatabaseTests
         [SetUp]
         public void RunBeforeAnyTests()
         {
-            var builder = new DbContextOptionsBuilder();
-            builder.UseNpgsql(ConnectionString.TestDatabase())
-                .UseSnakeCaseNamingConvention();
+            var builder = new DbContextOptionsBuilder<BonusCalcContext>();
+
+            var dataSourceBuilder = new NpgsqlDataSourceBuilder(ConnectionString.TestDatabase());
+            dataSourceBuilder.MapEnum<BandChangeDecision>();
+            var dataSource = dataSourceBuilder.Build();
+
+            builder
+                .UseNpgsql(dataSource)
+                .UseSnakeCaseNamingConvention()
+                .ConfigureWarnings(warnings =>
+                    warnings.Ignore(CoreEventId.ManyServiceProvidersCreatedWarning));
```

Comment:
Your tests run may have different configuration than main app startup.
Startup has the: `UseQuerySplittingBehavior`, while this doesn't.
Depending on defaults this may or may not lead tests run to behave differently than the actual app would.